### ISSUE TITLE
withCount() aggregate properties ($model->xxx_count) not resolved

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#510]
## Description

When using Eloquent's `withCount()`, Laravel adds magic properties with a `_count` suffix (e.g., `$label->contacts_count`). The plugin does not resolve these properties, causing `UndefinedMagicPropertyFetch` at Psalm errorLevel 1.

## Impact

**7 false positives** in [monicahq/monica](https://github.com/monicahq/monica) (262 project files). This is a common pattern in any Laravel app that uses aggregate loading.

Affected properties:
- `App\Models\Label::$contacts_count` (3 occurrences)
- `App\Models\Address::$contacts_count` (2)
- `App\Models\Vault::$contacts_count` (1)
- `App\Models\Tag::$posts_count` (1)

## Reproduction

Found while auditing [monicahq/monica](https://github.com/monicahq/monica) at Psalm errorLevel 1 with dev-master plugin.

```php
// In a controller/view helper:
$labels = Label::withCount('contacts')->get();

foreach ($labels as $label) {
    // UndefinedMagicPropertyFetch: Magic instance property App\Models\Label::$contacts_count is not defined
    $count = $label->contacts_count;
}
```

**File example:** `app/Domains/Contact/ManageContact/Web/ViewHelpers/ContactIndexViewHelper.php:32`

## Expected Behavior

The plugin should recognize that `withCount('relation')` adds an integer property `relation_count` to the model.

## Proposed Fix

The `ModelPropertyHandler` or `ModelRelationshipPropertyHandler` could recognize `_count` suffixed properties when the base name matches a defined relationship. The type would always be `int`.

Alternatively, this could be handled by recognizing `withCount()` calls in the query builder chain, but that's more complex. A simpler heuristic: if a model has a relationship `contacts()`, then `contacts_count` is a valid `int` property.
